src/passes/analysis/verilog.cpp

```diff
@@ -1116,18 +1105,36 @@ Passes::Verilog::compileModuleBody(
             wire_name,
             field_type,
             entries);
-
-        // Regular (possibly bulk) connection
-        body.push_back(std::make_unique<vAST::ContinuousAssign>(
-          std::move(target),
-          std::move(driver)));
+        auto id_ptr = dynamic_cast<vAST::Identifier*>(driver.get());
+        if (
+          id_ptr &&
+          std::holds_alternative<std::unique_ptr<vAST::Identifier>>(target)) {
+          // prevent inlining of this wire into the module instance statement
+          this->wires.insert(id_ptr->value);
+          verilog_connections->insert(field, std::move(driver));
+        }
+        else {
+          this->makeDecl(wire_name, field_type, body, false);
+          // prevent inlining of this wire into the module instance statement
+          this->wires.insert(wire_name);
+          verilog_connections->insert(
+            field,
+            std::make_unique<vAST::Identifier>(wire_name));
+
+          // Regular (possibly bulk) connection
+          body.push_back(std::make_unique<vAST::ContinuousAssign>(
+            std::move(target),
+            std::move(driver)));
+        }
       }
     }
     bool is_mem_inst = instance_module->isGenerated() &&
-        ((instance_module->getGenerator()->getName() == "mem" &&
-          instance_module->getGenerator()->getNamespace()->getName() == "coreir") ||
-         (instance_module->getGenerator()->getName() == "rom2" &&
-          instance_module->getGenerator()->getNamespace()->getName() == "memory"));
+      ((instance_module->getGenerator()->getName() == "mem" &&
+        instance_module->getGenerator()->getNamespace()->getName() ==
+          "coreir") ||
+       (instance_module->getGenerator()->getName() == "rom2" &&
+        instance_module->getGenerator()->getNamespace()->getName() ==
+          "memory"));
```

Comment:
Unrelated auto style change

Reply:
This is getting pretty big...can we put this in a helper function like `is_memory_primitive()`

Reply:
unrelated to the functionality of this change, but sure